src/tflite_cpu.cpp

```diff
@@ -0,0 +1,756 @@
+        // drain.
+    }
+
+    void stop(const vsdk::ProtoStruct& extra) noexcept final {
+        return stop();
+    }
+
+    /// @brief Stops the MLModelServiceTFLite from running.
+    void stop() noexcept {
+        using std::swap;
+        try {
+            const std::lock_guard<std::mutex> lock(state_lock_);
+            if (!stopped_) {
+                stopped_ = true;
+                std::shared_ptr<state> state;
+                swap(state_, state);
+                state_ready_.notify_all();
+            }
+        } catch (...) {
+        }
+    }
+
+    void reconfigure(const vsdk::Dependencies& dependencies,
+                     const vsdk::ResourceConfig& configuration) final try {
+        // Care needs to be taken during reconfiguration. The
+        // framework does not offer protection against invocation
+        // during reconfiguration. Keep all state in a shared_ptr
+        // managed block, and allow client invocations to act against
+        // current state while a new configuration is built, then swap
+        // in the new state. State which is in use by existing
+        // invocations will remain valid until the clients drain. If
+        // reconfiguration fails, the component will `stop`.
+
+        // Swap out the state_ member with nullptr. Existing
+        // invocations will continue to operate against the state they
+        // hold, and new invocations will block on the state becoming
+        // populated.
+        using std::swap;
+        std::shared_ptr<state> state;
+        {
+            // Wait until we have a state in play, then take
+            // ownership, so that we don't race with other
+            // reconfigurations and so other invocations wait on a new
+            // state.
+            std::unique_lock<std::mutex> lock(state_lock_);
+            state_ready_.wait(lock, [this]() { return (state_ != nullptr) && !stopped_; });
+            check_stopped_inlock_();
+            swap(state_, state);
+        }
+
+        state = reconfigure_(dependencies, configuration);
+
+        // Reconfiguration worked: put the state in under the lock,
+        // release the lock, and then notify any callers waiting on
+        // reconfiguration to complete.
+        {
+            std::lock_guard<std::mutex> lock(state_lock_);
+            check_stopped_inlock_();
+            swap(state_, state);
+        }
+        state_ready_.notify_all();
+    } catch (...) {
+        // If reconfiguration fails for any reason, become stopped and rethrow.
+        stop();
+        throw;
+    }
+
+    std::shared_ptr<named_tensor_views> infer(const named_tensor_views& inputs,
+                                              const vsdk::ProtoStruct& extra) final {
+        auto state = lease_state_();
+
+        // We serialize access to the interpreter. We use a
+        // unique_lock because we will move the lock into the shared
+        // state that we return, allowing the higher level to effect a
```

Comment:
Comment doesn't match code: we don't return anything. Leave as-is, and I'll fix it when you're on vacation.

Reply:
:facepalm: I've been doing too much Go recently, and looked for the return type near the word `final`. My mistake!